File: src/App/src/app/shared/models/group-details-dto.ts
export interface GroupMemberDto {
    id: string;
    name: string;
    role: number; // 0 = Owner, 1 = Manager, 2 = Member
}

export interface GroupInvitationDto {
    id: string;
    name: string;
    expirationDate: string;
}

export interface GroupJoinRequestDto {
    id: string;
    name: string;
    requestedAt: string;
}

export interface GroupDetailsDto {
    id: string;
    name: string;
    memberCount: number;
    isMember: boolean;
    isPublic: boolean;
    currentMemberRole: number | null; // 0 = Owner, 1 = Manager, 2 = Member, null if not a member
    members: GroupMemberDto[];
    invitations: GroupInvitationDto[];
    joinRequests: GroupJoinRequestDto[];
}

File: src/Groups/HexMaster.Attendr.Groups/GetGroupDetails/GroupDetailsDto.cs
namespace HexMaster.Attendr.Groups.GetGroupDetails;

/// <summary>
/// DTO representing detailed information about a group.
/// </summary>
/// <param name="Id">The unique identifier of the group.</param>
/// <param name="Name">The name of the group.</param>
/// <param name="MemberCount">The total number of members in the group.</param>
/// <param name="IsMember">Indicates whether the current profile is a member of the group.</param>
/// <param name="IsPublic">Indicates whether the group is public (anyone can join) or private (requires approval).</param>
public sealed record GroupDetailsDto(
    Guid Id,
    string Name,
    int MemberCount,
    bool IsMember,
    bool IsPublic);
